lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVMBase.h

```diff
@@ -1129,25 +1129,12 @@ class ConvertTritonGPUOpToLLVMPatternBase {
     formatString.push_back('\0'); // Null terminate for C
     size_t formatStringSize = formatString.size_in_bytes();
 
-    auto globalType = LLVM::LLVMArrayType::get(llvmI8, formatStringSize);
-    LLVM::GlobalOp global;
-    {
-      ConversionPatternRewriter::InsertionGuard guard(rewriter);
-      rewriter.setInsertionPointToStart(moduleOp.getBody());
-      global = rewriter.create<LLVM::GlobalOp>(
-          loc, globalType,
-          /*isConstant=*/true, LLVM::Linkage::Internal, stringConstName,
-          rewriter.getStringAttr(formatString));
-    }
+    Value prefixString =
+        LLVM::addStringToModule(loc, rewriter, "printfFormat_", formatString);
+
+    auto prefixPtrType = ocklAppendStringN.getArgumentTypes()[1];
+    prefixString = bitcast(prefixString, prefixPtrType);
```

Comment:
Need to cast, because `addStringToModule` returns ptr to i8, but `ocklAppendStringN` fucntion expects untyped ptr